src/pal/pal_posix.h

```diff
@@ -215,6 +215,24 @@ namespace snmalloc
         zero<true>(p, size);
     }
 
+    /**
+     * Notify platform that we will be using these pages for reading.
+     *
+     * On POSIX platforms, lazy commit means that this is a no-op, unless
+     * we have initially mapped the pages as PROT_NONE.
+     */
+    static void notify_using_readonly(void* p, size_t size) noexcept
+    {
+      SNMALLOC_ASSERT(is_aligned_block<OS::page_size>(p, size));
+
+#ifdef SNMALLOC_CHECK_CLIENT
```

Comment:
Conditioning on `SNMALLOC_CHECK_CLIENT` inside `notify_using_readonly` seems a little dodgy; at the moment the only call is already under that `ifdef` guard already, but I could imagine cases where we'd be wanting this not under that guard.

Reply:
Good point, at the moment the Pals behave differently for CHECK_CLIENT and not, with initially mapping memory readonly.  This isn't ideal. I think a larger refactor to make Pals not depend on CHECK_CLIENT makes sense, but then we will need to expose more features from the Pal interface.

Reply:
Will refactor based on Teams conversation, and add a second commit.

Reply:
Performed a bit of refactoring to make the semantics, hopefully, clearer.